src/sinks/amqp/encoder.rs

```diff
@@ -11,9 +12,17 @@ pub(super) struct AmqpEncoder {
 }
 
 impl encoding::Encoder<Event> for AmqpEncoder {
-    fn encode_input(&self, mut input: Event, writer: &mut dyn io::Write) -> io::Result<usize> {
+    fn encode_input(
+        &self,
+        mut input: Event,
+        writer: &mut dyn io::Write,
+    ) -> io::Result<(usize, GroupedCountByteSize)> {
         let mut body = BytesMut::new();
         self.transformer.transform(&mut input);
+
+        let mut byte_size = telemetry().create_request_count_byte_size();
+        byte_size.add_event(&input, input.estimated_json_encoded_size_of());
+
         let mut encoder = self.encoder.clone();
         encoder
             .encode(input, &mut body)
```

Comment:
This is a bit outside the scope of the PR, but I'm curious why we need to estimate the JSON size right before we actually encode the event. Don't we have the actual size here that we could use directly?

Reply:
This was part of https://github.com/vectordotdev/vector/issues/17576. The intention was to use a measurement that could be taken anywhere in the processing pipeline (source, transforms, sinks) so that the values are comparable.